rfcs/text/0010-email.md

```diff
@@ -1,68 +1,44 @@
+This RFC proposes a new top-level field set to facilitate email use cases, `email.*`. The `email.*` field set adds fields for the sender, recipient, message header fields, and other attributes of an email message typically seen logs produced by mail transfer agent (MTA) and email gateway applications.
 
 ## Fields
 
 <!--
 Stage 1: Describe at a high level how this change affects fields. Which fieldsets will be impacted? How many fields overall? Are we primarily adding fields, removing fields, or changing existing fields? The goal here is to understand the fundamental technical implications and likely extent of these changes. ~2-5 sentences.
 -->
 
-Email specific fields:
+### Email specific fields
 
 | field | type | description |
 | --- | --- | --- |
-| `email.bcc.addresses` | `wildcard[]` | Addresses of Bcc's |
-| `email.cc.addresses` | `wildcard[]` | Addresses of Cc's |
-| `email.attachments_count` | long | A field outside the flattened structure to control how many attachments are included in the email |
-| `email.attachments` | flattened | A flattened field for anything related to attachments. This allows objects being stored with all information for each file when you have multiple attachments |
-| `email.direction` | keyword | Direction of the message based on the sending and receving domains |
-| `email.sender.address` | wildcard | Senders email address |
-| `email.sender.domain` | wildcard | Domain of the sender |
-| `email.sender.top_level_domain` | keyword | Top level domain of the sender |
-| `email.sender.registered_domain` | wildcard | Registered domain of the sender |
-| `email.sender.subdomain` | keyword | Subdomain of the sender |
-| `email.message_id` | keyword | Internet message ID of the message |
-| `email.reply_to.address` | wildcard | Reply-to address |
-| `email.return_path.address` | wildcard | The return address for the message |
-| `email.size` | long | Total size of the message, in bytes, including attachments |
-| `email.subject` | wildcard | Subject of the message |
-| `email.recipients.addresses` | `keyword[]` | Recipient addresses |
-| `email.domains` | `keyword[]` | domains related to the email |
-
-
-Other ECS fields used together with email usecases:
-| field | description |
-| --- | --- |
-| `event.duration` | The duration related to the email event. Could be the total duration in Quarantine, how long the email took to send from source to destination etc |
-| `event.start` | When the email event started
-| `event.end` | When the email event ended
-| `process.name` | When the event is related to a server or client. Does not take MTA into account which is part of a ongoing discussion |
-| `network.protocol` | Type of email protocol used |
-| `tls.*` | Used for TLS related information for the connection to for example a SMTP server over TLS |
-
-
+| `email.from` | keyword (array) | Stores the `from` email address(es) |
+| `email.timestamp` | keyword (array) | The local date and time the message was written |
```

Comment:
I don't see a concept representing when an email was received, which is another time value useful for security purposes.

Reply:
This timestamp captures the [origination date](https://datatracker.ietf.org/doc/html/rfc5322#section-3.6.1) value. The description should explain that better; I'll address it.

I had considered using something like `email.sent_timestamp` and `email.received_timestamp`. I struggled with `received,` though.  A log emitted from an MTA (like Mimecast) might only have a single timestamp, but if the source is similar to an EML file with [trace fields](https://datatracker.ietf.org/doc/html/rfc5322#section-3.6.7). Open to any ideas on how to best represent these nuances.

cc @peasead

Reply:
Good question.

It looks like trace fields are part of a different [RFC](https://datatracker.ietf.org/doc/html/rfc5321#section-4.4), maybe those trace fields would be better tracked in an SMTP fieldset?

If not, would there be a way to make the field an array of timestamps if necessary? Is there a way to connect the different trace timestamps with the other trace elements (return, received, etc.)

Reply:
After some thought, I added an `email.delivery_timestamp` field described as `The date and time the email message was received by the service or client.`. This reserves the word `received` for potential use with the trace records (which aren't currently part of this proposal).

Thoughts?